device/common/include/traccc/finding/device/impl/apply_interaction.ipp

```diff
@@ -49,25 +36,16 @@ TRACCC_DEVICE inline void apply_interaction(
     auto& bound_param = params.at(globalIndex);
 
     // Get intersection at surface
-    const detray::surface<detector_t> sf{det, bound_param.surface_link()};
-    using cxt_t = typename detector_t::geometry_context;
-    const cxt_t ctx{};
-    const auto free_vec = sf.bound_to_free_vector(ctx, bound_param.vector());
-
-    const auto& mask_store = det.mask_store();
-    intersection_type sfi;
-    sfi.sf_desc = det.surface(bound_param.surface_link());
-    sf.template visit_mask<
-        detray::intersection_update<detray::ray_intersector>>(
-        detray::detail::ray<transform3_type>(free_vec), sfi,
-        det.transform_store());
+    const detray::surface sf{det, bound_param.surface_link()};
+    const typename detector_t::geometry_context ctx{};
 
     // Apply interactor
     typename interactor_type::state interactor_state;
     interactor_type{}.update(
         bound_param, interactor_state,
         static_cast<int>(detray::navigation::direction::e_forward), sf,
-        sfi.cos_incidence_angle);
+        std::abs(
+            sf.cos_angle(ctx, bound_param.dir(), bound_param.bound_local())));
```

Comment:
This should become [std::fabs](https://en.cppreference.com/w/cpp/numeric/math/fabs) for CUDA/HIP, and `cl::sycl::fabs` for SYCL. Though I don't know if we would have the infrastructure for this already. :thinking: